### PR TITLE
fix:OS-SWP-SUG-01

### PR DESCRIPTION
**Pull request by joaquinsoza**
Fixing audit review, issue OS-SWP-SUG-01 Checking if address are not equal

### CODE REVIEW

**Review by esteblock — CHANGES_REQUESTED** [review]
@joaquinsoza Please add an Error here, not a panic.
See src/error.rs.

**Comment by joaquinsoza** [comment]
Issue: #124 
Article: [Missing Address Check in Pair Contract](https://github.com/soroswap/articles/blob/main/audit-review/PairAddressCheck.md)

File: contracts/factory/src/lib.rs
Comment (esteblock) [inline contracts/factory/src/lib.rs:1]: I don't think that .map_err is correctly used here.
In my opinion, only the "?" should be fine,

**Review by esteblock — CHANGES_REQUESTED** [review]
Check the correct use of map_err

**Comment by esteblock** [comment]
This might work well.
Please check: https://github.com/soroswap/core/pull/146/commits/c34ef0a3fc7cf779fe3a8e80e6a4e63e94ef9932

File: contracts/factory/src/lib.rs
Comment (joaquinsoza) [reply contracts/factory/src/lib.rs:1]: Perfect, I had gone through a solution with "?" but I discarded it when I had to make many changes to the contract...

**Pull request merged**